PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,478 @@
+// -*- C++ -*-
+//
+// Package:    ProdTutorial/L1PrefiringWeightProducer
+// Class:      L1PrefiringWeightProducer
+//
+/**\class L1PrefiringWeightProducer L1PrefiringWeightProducer.cc ProdTutorial/L1ECALPrefiringWeightProducer/plugins/L1ECALPrefiringWeightProducer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  localusers user
+//         Created:  Thu, 08 Nov 2018 16:16:00 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+#include "TFile.h"
+#include "TF1.h"
+#include "TH2.h"
+
+#include <iostream>
+enum fluctuations { central = 0, up, down };
+
+class L1PrefiringWeightProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1PrefiringWeightProducer(const edm::ParameterSet&);
+  ~L1PrefiringWeightProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  double getPrefiringRate(double eta, double pt, TH2F* h_prefmap, fluctuations fluctuation) const;
+  double getPrefiringRateMuon(double eta, double phi, double pt, fluctuations fluctuation) const;
+
+  edm::EDGetTokenT<std::vector<pat::Photon> > photons_token_;
+  edm::EDGetTokenT<std::vector<pat::Jet> > jets_token_;
+  edm::EDGetTokenT<std::vector<pat::Muon> > muon_token_;
+
+  TF1* parametrization0p0To0p2_;
+  TF1* parametrization0p2To0p3_;
+  TF1* parametrization0p3To0p55_;
+  TF1* parametrization0p55To0p83_;
+  TF1* parametrization0p83To1p24_;
+  TF1* parametrization1p24To1p4_;
+  TF1* parametrization1p4To1p6_;
+  TF1* parametrization1p6To1p8_;
+  TF1* parametrization1p8To2p1_;
+  TF1* parametrization2p1To2p25_;
+  TF1* parametrization2p25To2p4_;
+  TF1* parametrizationHotSpot_;
+
+  TH2F* h_prefmap_photon;
+  TH2F* h_prefmap_jet;
+  std::string dataera_;
```

Comment:
What abot `dataeraEcal_` ?
They can also be made `const`

Reply:
Done.